spec/unit/auto_spin_spec.rb

```diff
@@ -12,12 +12,17 @@
     expect(spinner).to have_received(:spin).at_least(5).times
   end
 
-  it "restores cursor when erorr is raised" do
+  it "restores cursor when error is raised" do
     spinner = TTY::Spinner.new(output: output, hide_cursor: true)
 
-    spinner.auto_spin {
-      raise "boom"
-    }
+    Thread.report_on_exception = false
+    begin
+      spinner.run do
+        raise "boom"
+      end
+    rescue RuntimeError
+    end
+    Thread.report_on_exception = true
```

Comment:
This tests the `run` call rather than the `auto_spin`. I'd move this to the `spec/unit/run_spec.rb` file instead.

Reply:
`auto_spin` doesn't accept a block, so I'm really not sure what exactly the original test tried to check (and why it crashes at all). I also noticed that the cursor remains hidden even if the block is empty.